rust/private/rustc.bzl

```diff
@@ -716,7 +720,6 @@ def establish_cc_info(ctx, attr, crate_info, toolchain, cc_toolchain, feature_co
     link_input = cc_common.create_linker_input(
         owner = ctx.label,
         libraries = depset([library_to_link]),
-        user_link_flags = depset(toolchain.stdlib_linkflags),
```

Comment:
How do the linker flags get passed to the action with this omitted?

Reply:
They are added when the linker is actually run: https://github.com/bazelbuild/rules_rust/pull/955/files#diff-ecce7f6cdb4bc2f884f23fccb9a20174ad2b357dc6745314054e5eb8b784d2e0R222 (just above, you can see that code gathering the linker args from all of the dependencies)

Reply:
Sorry, the link there is hard to tell what line it's referring to. Could you link to the branch in your fork? 🙏 😅

Reply:
https://github.com/grafica/rules_rust/blob/deduplicate_platform_args/rust/private/rustc.bzl#L222

Reply:
Both of these links seem to highlight the exact same line via a fragment identifier, so, sorry if I'm being dense. Previously, the nested loop below this comment:

```
    # Add linkopt's from dependencies. This includes linkopts from transitive
    # dependencies since they get merged up.
```

duplicated these platform default link arguments. With this patch, they get added once at the end.

Reply:
Ah, I see now. Thanks for bearing with me while I learn to GitHub 😅 

This seems reasonable but I'm not sure off the top of my head what [cc_common.create_link_variables](https://docs.bazel.build/versions/4.0.0/skylark/lib/cc_common.html#create_link_variables) does in terms of propagating flags to upstream targets.

I think if you were to add some before+after evidence and a regression test that I'd be happy with this 😄

Reply:
> add some before+after evidence

With the patch, `-ldl -lpthread` appears once at the beginning of `--codegen 'link-args=...`. The status quo does this:

![Screen Shot 2021-09-25 at 12 46 41 PM](https://user-images.githubusercontent.com/253473/134784425-a3ae39c9-ba19-4118-8035-c78753f10cfb.png)

Reply:
OK, I added a test below. It only checks for deduplication on Linux, but it fails without this patch (in that case, the test targets insert three copies of the platform-default link flags).

I could add more platforms and flags, but I'm not sure how to fish around in ctx to get the information I would need under test.